src/table/utils/__tests__/handle-scroll.spec.ts

```diff
@@ -0,0 +1,199 @@
+      evt = {
+        ...evt,
+        deltaX: -1,
+      };
+      memoedContainer = {
+        ...memoedContainer,
+        scrollLeft: -100,
+      };
+      handleHorizontalScroll(evt, isRTL, memoedContainer);
+      expect(evt.stopPropagation).toHaveBeenCalledTimes(1);
+      expect(evt.preventDefault).toHaveBeenCalledTimes(1);
+    });
+
+    it('should run preventDefault when the scrollbar is at its rightmost place and and is scrolled right in RTL direction', () => {
+      evt = {
+        ...evt,
+        deltaX: 1,
+      };
+      memoedContainer = {
+        ...memoedContainer,
+        scrollLeft: 1,
+      };
+      handleHorizontalScroll(evt, isRTL, memoedContainer);
+      expect(evt.stopPropagation).toHaveBeenCalledTimes(1);
+      expect(evt.preventDefault).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not run preventDefault when the scrollbar is not at its leftmost place or the rightmost place in RTL direction', () => {
+      evt = {
+        ...evt,
+        deltaX: -10,
+      };
+      memoedContainer = {
+        ...memoedContainer,
+        scrollLeft: -50,
+      };
+      handleHorizontalScroll(evt, isRTL, memoedContainer);
+      expect(evt.stopPropagation).toHaveBeenCalledTimes(1);
+      expect(evt.preventDefault).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('handleNavigateTop', () => {
+    let tableContainerRef: {
+      current: {
+        scrollTo?: () => void;
+        scrollTop?: number;
+      };
+    };
+    let focusedCellCoord: [number, number];
+    let rootElement: Element;
+    let rowHeight: number;
+
+    beforeEach(() => {
+      rowHeight = 100;
+      tableContainerRef = {
+        current: {
+          scrollTo: jest.fn(),
+        },
+      };
+      focusedCellCoord = [0, 0];
+    });
+
+    it('should not do anything when ref is not setup yet', () => {
+      delete tableContainerRef.current.scrollTo;
+      expect(
+        handleNavigateTop({
+          tableContainerRef: tableContainerRef as React.MutableRefObject<HTMLDivElement>,
+          focusedCellCoord,
+          rootElement,
+        })
+      ).toBe(undefined);
+    });
+
+    it('should the scrollbar is at its top when you reach the top two rows', () => {
+      focusedCellCoord = [1, 0];
+
+      handleNavigateTop({
+        tableContainerRef: tableContainerRef as React.MutableRefObject<HTMLDivElement>,
+        focusedCellCoord,
+        rootElement,
+      });
+      expect(tableContainerRef.current.scrollTo).toHaveBeenCalledWith({ top: 0, behavior: 'smooth' });
+    });
+
+    it('should scroll upwards automatically if it detects the cursor gets behind <TableHead />', () => {
+      const SCROLL_TOP_IDX = 7;
+      focusedCellCoord = [8, 0];
+      tableContainerRef = {
+        current: {
+          scrollTo: jest.fn(),
+          scrollTop: SCROLL_TOP_IDX * rowHeight,
+        },
+      };
+
+      rootElement = {
+        getElementsByClassName: (query: string) => {
```

Comment:
There is an issue here: 
It complains about `getElementByClassName` which should be `(className: string) => <HTMLCollectionOfElement>`  but due to using a function(to set a value for height and top), the type has been changed. I tried to  kind of mock it or use casting but it didn't work!